src/test/java/com/bmdsoftware/pacs/dicoogle/ipfs/TestIPFS.java

```diff
@@ -22,68 +22,64 @@
 import io.ipfs.api.MerkleNode;
 import io.ipfs.api.NamedStreamable;
 import io.ipfs.multihash.Multihash;
-import org.junit.Ignore;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 
-public class TestIPFS {
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 
+/**
+ * @author Rui Lebre - <ruilebre@ua.pt>
+ */
+public class TestIPFS {
+    private static IPFS ipfs;
+    private static String testFileContent = "test test test";
 
+    @BeforeClass
+    public static void setup() {
+        ipfs = new IPFS("/ip4/127.0.0.1/tcp/5001");
```

Comment:
If the IPFS is not on, the tests will be ignored?

Reply:
Yes, bc if there is no connection with the IPFS daemon then the test itself cannot begin. So, the behaviour would be the unit test to be ignored and the test set to fail with `java.lang.RuntimeException: Couldn't connect to IPFS daemon at (...)`.

Do you think this is a bad practice? Maybe on erroneous behaviour on CI?

Reply:
Yes, but you still can warn about it. May be if behaviour of tests make sense to keep IPFS always up, and if not, we fail the tests. Another good possibility is to mock IPFS class using for instance mockito or plain mock.